src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueCreateDialog.kt

```diff
@@ -82,22 +84,46 @@ class IssueCreateDialog(
             openConfigureFieldsDialog(dialogLocator)
         }
 
-        val allFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-        if (driver.isElementPresent(allFieldsLinkLocator)) {
-            driver.wait(elementToBeClickable(allFieldsLinkLocator)).click()
-            val newAllFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-            driver.wait(not(elementToBeClickable(newAllFieldsLinkLocator)))
+        if (newConfigureFieldsUi) {
+            val selectLocator = getFieldsSelectLocator(configureFieldsDialogClass);
+
+            if (driver.isElementPresent(selectLocator)) {
```

Comment:
Can this condition become the differentiator?

Reply:
This is a good question, if this is true we could get rid of Jira's version comparison.

Reply:
Won't such fallbacks/decisions impact the measure? Shouldn't the decision be done in the action inself, before we start to measure? i.e. VU knows that we are on version X, so it uses page objects compatible with X

Reply:
Probably the measure will be affected a little bit tho it will be affected by the UI change itself. 

Also, I am not sure if I'll be able to access WebDriver object in the action itself. 

I am open to suggestions.

Reply:
I don't worry about the impact here, because:
* soon, most tests will short-circuit on the new GUI - the natural bias towards testing new versions will deepen with time
* a couple of WD calls is not a big cost - it's not an n+1
* if it distinguishes visually or structurally differences, then the human user also needs to spend time for recognition
    * the impact on VU might be too small actually
    * effectively, VUs using this code represent humans who are more familiar with new GUI and less familiar with old GUI
* we might drop support for old Jiras at some point, we can remove the fallbacks then